packages/web/lib/browser.js

```diff
@@ -119,30 +137,69 @@ const getBrowserScope = (parent) => ({
         });
       } catch (e) {}
 
-      if (match) {
-        // Generate a new page scope with the same page
-        const pageScope = getPageScope(scope);
-        // inject args and "url" as variables
-        Object.assign(pageScope.vars, match.path, {
-          url: href,
-          hash: match.hash,
-          query: match.query,
-        });
-
-        // Navigate to the page
-        pageScope.internal.page = await browser.newPage();
-        await pageScope.internal.page.goto(href);
-
-        // TODO: support multple matching definitions
-        await evalRuleSet(match.ruleSet, pageScope);
+      //Get the nearest browser scope
+      const nearestBrowserScope = resolveInternalScope("browser", scope);
 
-        // TODO: check if the url has changed? If so, recurse and execute the necessary `page` rule
-
-        // Finally, close the page
-        pageScope.internal.page.close();
+      //If the nearest browser scope doesn't have a browser, create one
+      let browser = nearestBrowserScope.internal.browser;
+      if (!browser) {
+        browser = nearestBrowserScope.internal.browser = await puppeteer.launch(
+          {
+            headless: false,
```

Comment:
hmm, I thought I updated this code on the latest `master`. The `launch` takes a few additional args and doesn't do headless anymore. Can you rebase on `master`